src/classes/Mfa.php

```diff
@@ -0,0 +1,178 @@
+<?php
+/**
+ * @package   Elabftw\Elabftw
+ * @author    Nicolas CARPi <nico-git@deltablot.email>
+ * @copyright 2012 Nicolas CARPi
+ * @license   https://www.gnu.org/licenses/agpl-3.0.html AGPL-3.0
+ * @see       https://www.elabftw.net Official website
+ */
+declare(strict_types=1);
+
+namespace Elabftw\Elabftw;
+
+use Elabftw\Services\Filter;
+use Elabftw\Services\MpdfQrProvider;
+use RobThree\Auth\TwoFactorAuth;
+use Symfony\Component\HttpFoundation\RedirectResponse;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Session\Session;
+use function time;
+
+/**
+ * Provide methods for multi/two-factor authentication
+ */
+class Mfa
+{
+    /** @var Session $Session Current session */
+    public $Session;
+
+    /** @var Db $Db SQL Database */
+    protected $Db;
+
+    /** @var TwoFactorAuth $TwoFactorAuth PHP Class for handling two/multi-factor authentication */
+    private $TwoFactorAuth;
+
+    /** @var Request $Request Current request */
+    private $Request;
+
+    /**
+     * Constructor
+     *
+     * @param Request $request
+     * @param Session $session
+     */
+    public function __construct(Request $request, Session $session)
+    {
+        $this->TwoFactorAuth = new TwoFactorAuth('eLabFTW', 6, 30, 'sha1', new MpdfQrProvider());
```

Comment:
let's use sha256 instead. Everybody is moving away from sha1

Reply:
[It seems to be a problem](https://labanskoller.se/blog/2019/07/11/many-common-mobile-authenticator-apps-accept-qr-codes-for-modes-they-dont-support/) to use something else than the 'default' (HMAC-SHA-1 algorithm, 6 digits and a period of 30 seconds) if you want to have a broad range of authenticator apps working. There is a [recommendation to use default values](https://labanskoller.se/blog/2019/07/11/many-common-mobile-authenticator-apps-accept-qr-codes-for-modes-they-dont-support/#recommendation-to-site-owners). In addition, the token does not need to be extremely secure as a potential attacker also needs the password and will be banned after some tries due to the anti brute force system.

On the other hand, eLabFTW could recommend an app that works with sha256. However, users might try to use an app they already have installed anyways and potentially will complain about 2FA not working on eLabFTW.

Reply:
Ok, let's keep sha1 then!